Refactor and Optimize Codebase for latest ComfyUI

toxicwind opened the pull request:
This pull request aims to enhance the codebase by improving code organization, readability, and maintainability. The changes made are as follows:

1. **Code Organization and Style:**
   - Sorted imported modules alphabetically.
   - Added type hints for function parameters and return values.
   - Removed unnecessary imports for code cleanliness.
   - Formatted the code according to PEP 8 style guidelines.

2. **Refactored Functions:**
   - Refactored functions like `adjust_tensor_shape`, `expand`, and `bounded_modulo` for improved efficiency and readability.

3. **`calc_cond` Function:**
   - Simplified and improved readability using dictionary comprehension.
   - Enhanced the code for determining 'orig_len' and selecting conditions.

4. **`catenate_conds` and `subscript_cond` Functions:**
   - Simplified and made more concise for better readability.

5. **`pad_cond` Function:**
   - Modified to use `F.pad` for efficient tensor padding.

6. **`CFGDenoiser` Class:**
   - Introduced methods like `prepare_inputs`, `prepare_cond`, `apply_model`, and `post_process` for modularity and readability.
   - Eliminated redundant code segments for conciseness and clarity.

7. **`CFGNoisePredictor` Class:**
   - Refactored the `apply_model` method for readability and modularity.
   - Introduced helper methods for efficient condition preparation.
   - Made the `txt2img_image_conditioning` method static for better organization.

8. **`SDKSampler` Class:**
   - Introduced the `set_model_wrap` method to set the `model_wrap` attribute for improved organization.
   - Modified the constructor to use the `set_model_wrap` method.

These changes aim to improve code quality and make it more readable and maintainable while preserving the original functionality of the code. The code remains functionally intact, but the enhancements result in a cleaner, more organized, and better-documented codebase.

jags111 committed:
Update README.md

Basic info on Comfy Efficiency Nodes.

jags111 committed:
Create readme.md

readme

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Create FUNDING.yml

funding options

jags111 committed:
Create readme.md

jags111 committed:
Add files via upload

all node images updated

jags111 committed:
Update README.md

jags111 committed:
Update bnk_adv_encode.py

changes to how advanced encode work .. possibly resolves the token issues

jags111 committed:
Add files via upload

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Add files via upload

jags111 committed:
Update README.md

jags111 committed:
Add files via upload

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update requirements.txt

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Add files via upload

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update efficiency_nodes.py

3 lines -4118,4119, 4121 changed

jags111 committed:
Update bnk_adv_encode.py

many small codes in the l is tokenized is modified

jags111 committed:
Update bnk_tiled_samplers.py

many changes in params and conditions

jags111 committed:
Add files via upload

jags111 committed:
Update efficiency_nodes.py

jags111 committed:
Update bnk_adv_encode.py

jags111 committed:
Update bnk_adv_encode.py

jags111 committed:
Update bnk_adv_encode.py

jags111 committed:
Update efficiency_nodes.py

jags111 committed:
Update bnk_adv_encode.py

jags111 committed:
Update smZ_cfg_denoiser.py

jags111 committed:
Update smZ_cfg_denoiser.py

jags111 committed:
Update bnk_tiled_samplers.py

jags111 committed:
Add files via upload

dnl13 committed:
Tiled Upscaler Script - removed controlnet model restriction

dnl13 committed:
fixes sdxl clip_l

jags111 committed:
Add files via upload

jags111 committed:
Update README.md

jags111 committed:
Add files via upload

dnl13 committed:
Merge remote-tracking branch 'upstream/main'

dnl13 committed:
added fix for smZ_cfg_denoiser by zdouble https://github.com/jags111/efficiency-nodes-comfyui/pull/12

jags111 committed:
Merge pull request #9 from dnl13/main

fix for tiles script issue

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Create funding.yml

ltdrdata committed:
fix: prevent the accumulation of model patches when 'xy_capsule' is used only for 'x'

jags111 committed:
Merge pull request #16 from ltdrdata/fix/xy_capsule

fix: prevent the accumulation of model patches when 'xy_capsule' is used only for 'x'

toxicwind committed:
Refactor tensor operations and streamline model application

- Implement efficient tensor shape adjustment for model inputs
- Optimize condition preparation and expansion in CFGNoisePredictor
- Ensure proper usage of tuples in tensor operations
- Enhance SDKSampler with dynamic model wrapping based on parameterization
- Remove redundant code and improve readability and performance

toxicwind committed:
Refactor tensor operations and streamline model application

- Implement efficient tensor shape adjustment for model inputs
- Optimize condition preparation and expansion in CFGNoisePredictor
- Ensure proper usage of tuples in tensor operations
- Enhance SDKSampler with dynamic model wrapping based on parameterization
- Remove redundant code and improve readability and performance

toxicwind committed:
gitignore

toxicwind committed:
.

toxicwind committed:
Merge branch 'main' of https://github.com/toxicwind/efficiency-nodes-comfyui-updates

toxicwind committed:
Remove pycache

jags111 committed:
Update FUNDING.yml

h3clikejava commented:
ERROR:root:!!! Exception during processing !!!
ERROR:root:Traceback (most recent call last):
  File "/home/h3c/Documents/ComfyUI/execution.py", line 153, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "/home/h3c/Documents/ComfyUI/execution.py", line 83, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "/home/h3c/Documents/ComfyUI/execution.py", line 76, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "/home/h3c/Documents/ComfyUI/custom_nodes/efficiency-nodes-comfyui/efficiency_nodes.py", line 702, in sample
    samples, images, gifs, preview = process_latent_image(model, seed, steps, cfg, sampler_name, scheduler,
  File "/home/h3c/Documents/ComfyUI/custom_nodes/efficiency-nodes-comfyui/efficiency_nodes.py", line 505, in process_latent_image
    positive_hash = tensor_to_hash(positive[0][0])
  File "/home/h3c/Documents/ComfyUI/custom_nodes/efficiency-nodes-comfyui/tsc_utils.py", line 96, in tensor_to_hash
    byte_repr = tensor.cpu().numpy().tobytes()  # Convert tensor to bytes
AttributeError: 'NoneType' object has no attribute 'cpu'

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Add files via upload

spinagon committed:
Fix "Fit" overlay resize mode

spinagon committed:
For some image sizes there would be a 1 pixel gap due to rounding

jags111 committed:
Merge pull request #24 from spinagon/main

Fix "Fit" overlay resize mode

jags111 committed:
Update efficiency_nodes.py

jags111 committed:
Update readme.md

jags111 committed:
Rename readme.md to README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update workflowfix.js

jags111 committed:
Update workflowfix.js

jags111 committed:
Update README.md

jags111 committed:
Update bnk_tiled_samplers.py

memory management changed in comfy original repo so the code got changed.

jags111 committed:
Update README.md

jags111 committed:
Update bnk_tiled_samplers.py

jags111 committed:
Update city96_latent_upscaler.py

added a line on noise mask

jags111 committed:
Create readme.md

for adding some latent models

jags111 committed:
Update README.md

models folder instructions added

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update __init__.py

jags111 committed:
Update __init__.py

jags111 committed:
Update city96_latent_upscaler.py

jags111 committed:
Update ttl_nn_latent_upscaler.py

jags111 committed:
Merge pull request #44 from jags111/jags111-patch-1

Update ttl_nn_latent_upscaler.py

jags111 committed:
Update __init__.py

jags111 committed:
Update __init__.py

jags111 committed:
Update __init__.py

KarryCharon committed:
fix #46

jags111 committed:
Merge pull request #47 from KarryCharon/fix-#46

fix #46

jags111 committed:
Revert "fix #46"

jags111 committed:
Merge pull request #54 from jags111/revert-47-fix-#46

Revert 47 fix #46

jags111 committed:
Add files via upload

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update appearance.js

jags111 committed:
Update seedcontrol.js

jags111 committed:
Update widgethider.js

jags111 committed:
Update ttl_nn_latent_upscaler.py

jags111 committed:
Update city96_latent_upscaler.py

jags111 committed:
Update README.md

jags111 committed:
Update README.md

jags111 committed:
Update README.md

alexopus committed:
Add LoRA Stack to String converter

jags111 committed:
Merge pull request #81 from alexopus/lora_stack_to_string

Add LoRA Stack to String converter

jags111 committed:
Add files via upload

nidefawl committed:
Fix add_noise and return_with_leftover_noise parameters

KSamplerAdvanced takes "enable" and "disable" strings instead of boolean for these parameters

jags111 committed:
Update efficiency_nodes.py

jags111 committed:
Update README.md

jags111 committed:
Update bnk_adv_encode.py

slouffka committed:
Fix CFG denoiser.

idrirap committed:
feat: allow to use negative and positive prompt S/R both in X and Y

idrirap committed:
feat: allow to use both pixel upscale and hires generation steps from latent

jags111 committed:
Merge pull request #89 from nidefawl/fix-add-noise-return-noise

Fix add_noise and return_with_leftover_noise parameters

jags111 committed:
Merge pull request #132 from idrirap/dev/better_pixel_upscale

feat: allow to use both pixel upscale and hires generation steps fromlatent

jags111 committed:
Merge pull request #130 from idrirap/dev/allowSR_XY

feat: allow to use negative and positive prompt S/R both in X and Y

jags111 committed:
Merge pull request #127 from slouffka/main

Fix CFG denoiser.

jags111 committed:
Update smZ_cfg_denoiser.py

jags111 committed:
Update smZ_cfg_denoiser.py

jags111 committed:
Update smZ_cfg_denoiser.py

jags111 committed:
Update smZ_cfg_denoiser.py

jags111 committed:
Update smZ_cfg_denoiser.py

haohaocreates committed:
chore(publish): $'Add Github Action for Publishing to Comfy Registry'

Commit:
chore(pyproject): $'Add pyproject.toml for Custom Node Registry'

jags111 committed:
Merge pull request #181 from haohaocreates/publish

Add Github Action for Publishing to Comfy Registry

jags111 committed:
Merge pull request #182 from haohaocreates/pyproject

Add pyproject.toml for Custom Node Registry

jags111 committed:
Update pyproject.toml

jags111 committed:
Add files via upload

jags111 committed:
Update pyproject.toml

jags111 committed:
Update pyproject.toml

jags111 committed:
Update pyproject.toml

jags111 committed:
Update pyproject.toml

jags111 committed:
Update pyproject.toml

Shinihime committed:
Load settings fix

"both" was missing

Shinihime committed:
Show latent upscaler when "both" is selected

jags111 committed:
Merge pull request #183 from Shinihime/patch-1

Load settings fix

jags111 committed:
Merge pull request #184 from Shinihime/patch-2

Show latent upscaler when "both" is selected

jags111 committed:
Update bnk_tiled_samplers.py

omfy.sampler's 'convert cond' has been removed so line 127 changes

jags111 committed:
Update bnk_tiled_samplers.py

jags111 committed:
Update bnk_tiled_samplers.py

changes in line 131 check same

jags111 committed:
Update pyproject.toml

jags111 committed:
Update bnk_tiled_samplers.py

 module 'comfy.sample' has no attribute 'get_additional_models'

jags111 committed:
Update bnk_tiled_samplers.py

jags111 committed:
Update bnk_tiled_samplers.py

rgthree committed:
Ensure to check a widget's prototype for `value` property descriptor. (#203)

This fixes cases where the widget is not a plain object, but a class instance with `value` getters/setters.

larsupb committed:
Bugfix encode step for XY_Capsule (inspire pack) (#187)

* bugfix: in case of XY_Capsule, prompt needs to be encoded by the analysed LoRA's clip

* bugfix: when processing LoRAs you need to reload the LoRA every index on the y scale (not only the first one) to maintain the LoRA-patched clip.

* add align your step schedulers via monkey patching

---------

Co-authored-by: larsupb <larsupb@gmx.net>

jags111 committed:
Update pyproject.toml

toxicwind committed:
.

toxicwind committed:
Merge branch 'main' of https://github.com/toxicwind/efficiency-nodes-comfyui-updates
